src/main/java/io/github/mweirauch/micrometer/jvm/extras/ContextSwitchMetrics.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2017-2022 Michael Weirauch (michael.weirauch@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.mweirauch.micrometer.jvm.extras;
+
+import io.github.mweirauch.micrometer.jvm.extras.procfs.ProcfsStatus;
+import io.github.mweirauch.micrometer.jvm.extras.procfs.ProcfsStatus.KEY;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import java.util.Objects;
+
+public class ContextSwitchMetrics implements MeterBinder {
+
+    private final ProcfsStatus status;
+
+    public ContextSwitchMetrics() {
+        this(ProcfsStatus.getInstance());
+    }
+
+    /* default */ ContextSwitchMetrics(ProcfsStatus status) {
+        this.status = Objects.requireNonNull(status);
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        FunctionCounter.builder("process.voluntary.context.switch", status, statusRef -> value(KEY.VOLUNTARY_CTXT_SWITCHES))
```

Comment:
To keep the "reverse-notation": `process.threads.context.switches.voluntary`? (Same would apply to the non voluntary ones.)

Reply:
good idea applied